llvm/lib/CodeGen/WindowScheduler.cpp

```diff
@@ -198,6 +198,12 @@ bool WindowScheduler::initialize() {
     if (MI.isMetaInstruction() || MI.isTerminator())
       continue;
     if (MI.isPHI()) {
+      if (Register AntiReg = getAntiRegister(&MI))
+        // Register with Kernel in phi is not defined within the Kernel itself.
+        if (MRI->getVRegDef(AntiReg)->getParent() != MBB) {
+          LLVM_DEBUG(dbgs() << "Special phi structure is not supported!\n");
```

Comment:
It's not really special. Why is this difficult to handle? It should behave like any other live in value

Reply:
Yes, you are right. We only need to add a small amount of checks.
Updated.